solver.rb

```diff
@@ -0,0 +1,27 @@
+class Solver
+  def factorial(num)
+    if num >= 0
+      return 1 if num.zero?
+
+      answer = num
+      until num == 1
+        num -= 1
+        answer *= num
+      end
+      return answer
+    end
+    'Not a valid number'
```

Comment:
- [ ] if the value is negative an error should be raised, kindly implement this instead of returning `'Not a valid number'`